shared/Cargo.toml

```diff
@@ -29,6 +29,7 @@ num-bigint = "0.3"
 once_cell = "1.8.0"
 primitive-types = "0.9"
 prometheus = "0.12"
+prometheus-metric-storage = "0.2"
```

Comment:
I see this is your crate, nice.

Reply:
If you could take a look at the code, I'd appreciate it: https://github.com/taminomara/prometheus-metric-storage

Reply:
Specifically I'd like to discuss implementation of this function: https://github.com/taminomara/prometheus-metric-storage/blob/main/prometheus-metric-storage/src/lib.rs#L120

I don't know how to do this without an unsafe block.

We could change signature to return `Result<T>` (so, no reference), but then we'll have to clone a bunch of metric storages and save them in our structs. For example, [this field here](https://github.com/gnosis/gp-v2-services/pull/1100/files#diff-7624483e59540f6a621fdf505941ddab387cf0b3fd3673dceb537fddc5411380R26) will have type `TransportMetrics` instead of `&'static TransportMetrics`. And then we'll have to clone metrics [here](https://github.com/gnosis/gp-v2-services/pull/1100/files#diff-7624483e59540f6a621fdf505941ddab387cf0b3fd3673dceb537fddc5411380R131) to avoid capturing `self` into the async block. Overall, this will cause lots of small inconveniences.